R/calc_AUC.R

```diff
@@ -0,0 +1,99 @@
+require(caTools)
+
+calc_AUC <-
+  function(qp,
+           test_matrix_Xi,
+           test_matrix_Xip,
+           test_yi) {
+    rankdiff <- function(qp, matrix_Xi, matrix_Xip, tau) {
+      Xirank <- qp$rank.scaled(X = matrix_Xi)
+      Xiprank <- qp$rank.scaled(X = matrix_Xip)
+      Xirank <- cbind(Xirank, Xiprank)
+      
+      Xirank <- data.frame(Xirank)
+      
+      for (i in 1:nrow(Xirank)) {
+        diff <- Xirank[i, 2] - Xirank[i, 1]
+        if (diff < -1 * tau) {
+          Xirank[i, 3] <- -1
+        }
+        if (diff > tau) {
+          Xirank[i, 3] <- 1
+        }
+        if (abs(diff) <= tau) {
+          Xirank[i, 3] <- 0
+        }
+      }
+      return(Xirank)
+    }
+    
+    out_AUC <- data.frame(tau = 0,
+                          TP = 0,
+                          FP = 0,
+                          i = 0)
+    
+    rankdiff_out <-
+      rankdiff(qp, test_matrix_Xi, test_matrix_Xip,tau = 1)
+    rankdiff_out$difference <- abs(rankdiff_out$X2 - rankdiff_out$X1)
+    
+    tau_array <- c(abs(rankdiff_out$difference))
+    
+    out_AUC_SVM_list  <- list()
+    
+    array_counter <- 1
+    
+    for (i in tau_array) {
```

Comment:
this for loop is OK in this PR, but for speed improvements (probably a future PR), I would suggest that you start thinking about how to implement the ROC curve computation without a for loop, using vectorized functions, see https://github.com/tdhock/WeightedROC/blob/master/R/ROC.R for an example of how to compute ROC curves for usual binary classification without for loops.